inference-engine/include/ie_blob.h

```diff
@@ -231,6 +264,20 @@ class INFERENCE_ENGINE_API_CLASS(Blob) {
      */
     virtual void* getHandle() const noexcept = 0;
 
+    /**
+     * @brief Clones Blob object of the particular class. Clone references the same memory as this object.
+     *
+     * @return Pointer to the cloned object
+     */
+    virtual Blob* clone() const = 0;
```

Comment:
Is copy assignment operator is available along with `clone`?

Reply:
not necessarily; it's not required to be present to solve the task of cloning

Reply:
I'm just curious about interface. If there are copy assignment operator and `clone` at the same time, I think it worthies to make it clear which one is deep and which is not. If `clone` is *shallow* , did you consider reflecting it in the name of method like `shallowClone` or something?

Reply:
`clone()` description tells so.
Are you on your own when thinking about another name?

Reply:
Didn't get your question

Reply:
Have you come up with the idea about the name by yourself?

Reply:
Yes
`clone` is a bit confusing to me because there is common pattern `clone` in C++ that resolves slicing problem while copying the interface (having pointer to base copy the actual subclass, not just base) and it seems to me it is about deep copy, but I may be wrong

Reply:
This `clone` is indeed about copying the state of derived object by having only base* pointer.
Since you are not the first who is curious for another name, it can be voted

Reply:
Vote:
`shallowClone`

Reply:
Vote:
`clone`

Reply:
`shallowClone` 👍